sort.rb

```diff
@@ -0,0 +1,25 @@
+
+def sort(array) 
+  if array == nil || array.length <= 1
+    return array # nothing to sort
+  end
+
+  index = 0
+  swapped = true
+  while index < array.length && swapped
+    swapped = false
+    other_index = index + 1
+    while other_index < array.length
+      if array[index] > array[other_index]
```

Comment:
This block reads cohesively to me; however, I wonder how much the nesting slows its performance. Is there a way we could write it with less nesting to optimize speed?